src/tests/suite/daos_rebuild.c

```diff
@@ -1115,9 +1115,13 @@ rebuild_fail_all_replicas_before_rebuild(void **state)
 	print_message("sleep 10 seconds to wait scan to be finished \n");
 	sleep(10);
 
-	/* Then kill rank 1 */
-	daos_kill_server(arg, arg->pool.pool_uuid, arg->group,
-			 arg->pool.alive_svc, shard->os_ranks[1]);
+	/* Then kill rank on shard1 */
+	/* NB: we can not kill rank 0, otherwise the following set_params
+	 * will fail and also pool destory will not work.
+	 */
+	if (shard->os_ranks[1] != 0)
+		daos_kill_server(arg, arg->pool.pool_uuid, arg->group,
```

Comment:
do you not still need to kill a server even if it's not rank 0 ?

Reply:
rank 0 is chosen randomly. If rank 0 is picked here, then it will not skip this kill, and only kill 1 rank.